File: basic-samples/src/main/java/com/github/ljtfreitas/julian/samples/basic/Callbacks.java
package com.github.ljtfreitas.julian.samples.basic;

import com.github.ljtfreitas.julian.Header;
import com.github.ljtfreitas.julian.Headers;
import com.github.ljtfreitas.julian.ProxyBuilder;

public class Callbacks {

    public static void main(String[] args) throws InterruptedException {
        HTTPBin httpBin = new ProxyBuilder().build(HTTPBin.class, "https://httpbin.org");

        httpBin.get(result -> {
            System.out.println("GET: " + result);
            System.out.println("#############");
        });

        httpBin.post("hello", result -> {
            System.out.println("POST: " + result);
            System.out.println("#############");
        });

        httpBin.put("hello", result -> {
            System.out.println("PUT:" + result);
            System.out.println("#############");
        });

        httpBin.delete(result -> {
            System.out.println("DELETE: " + result);
            System.out.println("#############");
        });

        httpBin.headers(Headers.create(
                new Header("x-my-header", "whatever"),
                new Header("x-my-other-header", "whatever")
        ), result -> {
            System.out.println("Headers: " + result);
            System.out.println("#############");
        });

        Thread.sleep(2000);
    }
}

File: kotlin-samples/src/main/kotlin/com/github/ljtfreitas/julian/samples/kotlin/Callbacks.kt
package com.github.ljtfreitas.julian.samples.kotlin

import com.github.ljtfreitas.julian.k.KFunctionCallbackResponseT
import com.github.ljtfreitas.julian.k.proxy
import com.github.ljtfreitas.julian.k.responses

fun main(args: Array<String>) {
    val httpBin = proxy<HTTPBin>(endpoint = "https://httpbin.org") {
        responses {
            add(KFunctionCallbackResponseT)
        }
    }

    httpBin.get { result ->
        println("GET: $result")
        println("#############")
    }

    httpBin.post("hello") { result ->
        println("POST: $result")
        println("#############")
    }

    httpBin.put("hello") { result ->
        println("PUT: $result")
        println("#############")
    }

    httpBin.delete { result ->
        println("DELETE: $result")
        println("#############")
    }

    Thread.sleep(2000)
}
